portfolio/src/main/java/com/google/sps/servlets/GameServlet.java

```diff
@@ -0,0 +1,172 @@
+    private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Scores of games currently in progress.
+    private HashMap<Long, Integer> curScores = new HashMap<>();
+
+    // Next answers for games currently in progress.
+    private HashMap<Long, Integer> ans = new HashMap<>();
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        // Load request params into strings.
+        String startTimeStr = request.getParameter("start");
+        String userAnsStr = request.getParameter("ans");
+        String name = request.getParameter("name");
+
+        // A post request must have a game start time and either an answer or username (for the highscore list).
+        if (startTimeStr == null || (userAnsStr == null && name == null)) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return;
+        }
+
+        long startTime = Long.parseLong(startTimeStr);
+        if (userAnsStr != null) {
+
+            // Error if the game is over or if the start time is in the future (modulo some error).
+            // Prevents hacking via extending game duration.
+            long time = System.currentTimeMillis();
+            if (time - startTime > 60500 || time - startTime < -50) {
+                response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+                return;
+            }
+
+            try {
+
+                // Update the users score (they are identified by their game start time).
+                int userAns = Integer.parseInt(userAnsStr);
+                int correct = 0;
+                if (userAns == ans.getOrDefault(startTime, Integer.MIN_VALUE)) {
+                    correct = 1;
+                }
+                curScores.put(startTime, curScores.getOrDefault(startTime, 0) + correct);
+            } catch (NumberFormatException e) {
+                // If they didn't enter a number, no need to update scores.
+            }
+
+        // If name param exists, this is a request to be added to the highscore list.
+        } else if (name != null) {
+
+            // Create new entity on highscore list.
+            Entity scoreEntity = new Entity("score");
+            scoreEntity.setProperty("name", name);
+            scoreEntity.setProperty("score", curScores.getOrDefault(startTime, 0));
+            datastore.put(scoreEntity);
+
+            // Remove entries from HashMaps to keep them small.
+            curScores.remove(startTime);
+            ans.remove(startTime);
+        }
+
+        // Send back the user's updated score.
+        response.setContentType("text/plain");
+        response.getWriter().println(curScores.getOrDefault(startTime, 0));
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        String fontConfig = System.getProperty("java.home")
```

Comment:
Please add a comment explaining why these hardcoded property gets.

Reply:
Added a description. This is due to a strange bug with appengine referenced here:
https://stackoverflow.com/questions/45100138/how-to-configure-google-appengine-to-work-with-vector-graphic